pkg/store/proxy_test.go

```diff
@@ -42,16 +43,19 @@ func (c *testClient) String() string {
 	return "test"
 }
 
+func (c *testClient) Addr() string {
+	return "testaddr"
+}
 func TestProxyStore_Info(t *testing.T) {
 	defer leaktest.CheckTimeout(t, 10*time.Second)()
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	q := NewProxyStore(nil,
+	q := NewProxyStore(nil, prometheus.NewRegistry(),
 		func() []Client { return nil },
 		component.Query,
-		nil,
+		nil, 1*time.Second,
```

Comment:
mhm, so 1 second will cause the CI flakiness. Be prepared that CI can be super slow. 

That's why time-based tests are quite difficult to implement with slow CI. We might want to:
* either ensure big enough timeout or no timeout at all for normal test cases
* inject `time.Now()` or `custom context implementation` to manually control this.

Reply:
I am afraid of the PRs being blocked by this once CircleCI is slow....